fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/electric/objects/ChargingStationObject.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.fed.application.ambassador.simulation.electric.objects;
+
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.objects.PointBoundingBox;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.objects.SpatialObject;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.objects.SpatialObjectBoundingBox;
+import org.eclipse.mosaic.lib.geo.CartesianPoint;
+import org.eclipse.mosaic.lib.objects.electricity.ChargingStationData;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+public class ChargingStationObject extends SpatialObject<ChargingStationObject> {
+    /**
+     * The data object that stores all static and dynamic information of the charging station.
+     */
+    private ChargingStationData chargingStationData;
+
+    /**
+     * The bounding box of a charging station is represented by a single point.
+     */
+    private transient PointBoundingBox boundingBox;
+
+    public ChargingStationObject(String id) {
+        super(id);
+    }
+
+
+    public ChargingStationObject setChargingStationData(ChargingStationData chargingStationData) {
+        this.chargingStationData = chargingStationData;
+        return this;
+    }
+
+    public ChargingStationData getChargingStationData() {
+        return chargingStationData;
+    }
+
+    @Override
+    public ChargingStationObject setPosition(CartesianPoint position) {
+        cartesianPosition.set(position);
+        position.toVector3d(this);
+        return this;
+    }
+
+    @Override
+    public SpatialObjectBoundingBox getBoundingBox() {
+        if (boundingBox == null) {
+            boundingBox = new PointBoundingBox(this);
+        }
+        return boundingBox;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ChargingStationObject that = (ChargingStationObject) o;
+
+        return new EqualsBuilder()
+                .appendSuper(super.equals(o))
+                .append(chargingStationData, that.chargingStationData)
+                .append(boundingBox, that.boundingBox)
+                .isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(5, 11)
+                .appendSuper(super.hashCode())
+                .append(chargingStationData)
```

Comment:
ChargingStationData seems not to override `hashCode` and `equals`. That must be added for this code to work properly.